src/lsdb/loaders/hats/read_hats.py

```diff
@@ -146,17 +146,26 @@ def _load_object_catalog(hc_catalog, config):
     catalog = Catalog(dask_df, dask_df_pixel_map, hc_catalog)
     if config.search_filter is not None:
         catalog = catalog.search(config.search_filter)
-    if isinstance(config.margin_cache, MarginCatalog):
-        catalog.margin = config.margin_cache
-        if config.search_filter is not None:
-            # pylint: disable=protected-access
-            catalog.margin = catalog.margin.search(config.search_filter)
-    elif config.margin_cache is not None:
-        hc_catalog = hc.read_hats(config.margin_cache)
-        catalog.margin = _load_margin_catalog(hc_catalog, config)
+    if config.margin_cache is not None:
+        margin_hc_catalog = hc.read_hats(config.margin_cache)
+        margin = _load_margin_catalog(margin_hc_catalog, config)
+        _validate_margin_catalog(margin_hc_catalog, hc_catalog)
+        catalog.margin = margin
     return catalog
 
 
+def _validate_margin_catalog(margin_hc_catalog, hc_catalog):
+    """Validate that the margin catalog and the main catalog are compatible"""
+    pixel_columns = [paths.PARTITION_ORDER, paths.PARTITION_DIR, paths.PARTITION_PIXEL]
+    margin_pixel_columns = pixel_columns + ["margin_" + column for column in pixel_columns]
+    catalog_schema = pa.schema([field for field in hc_catalog.schema if field.name not in pixel_columns])
+    margin_schema = pa.schema(
+        [field for field in margin_hc_catalog.schema if field.name not in margin_pixel_columns]
+    )
+    if not catalog_schema.equals(margin_schema):
```

Comment:
Can we exercise this in test? e.g. with 

`lsdb.read_hats(small_sky_object, margin=small_sky_source_margin)`